tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -2965,15 +3144,18 @@ private[optimizer] abstract class OptimizerCore(
   }
 
   private def fold3WayComparison(canBeEqual: Boolean, canBeLessThan: Boolean,
-      canBeGreaterThan: Boolean, lhs: Tree, rhs: Tree)(
-      implicit pos: Position): Tree = {
+      canBeGreaterThan: Boolean, lhs: PreTransform, rhs: PreTransform)(
+      implicit pos: Position): PreTransform = {
     import BinaryOp._
     if (canBeEqual) {
       if (canBeLessThan) {
-        if (canBeGreaterThan)
-          Block(keepOnlySideEffects(lhs), keepOnlySideEffects(rhs), BooleanLiteral(true))
-        else
+        if (canBeGreaterThan) {
+          PreTransBlock(
+              Block(finishTransformStat(lhs), finishTransformStat(rhs)),
+              PreTransLit(BooleanLiteral(true)))
```

Comment:
Your first suggestion would work (if we enhance the constructor of `PreTransBlock(List[PreTransBinding])` to get rid of `Right(Skip())`s eagerly) but I am not sure why it is better than the above?

That said, now I realize that what I wrote is stupid. Since `PreTransLit` gives a `PreTransTree`, that constructor of `PreTransBlock` will turn everything into a `PreTransTree`, and so I should simply write:
```scala
PreTransTree(Block(
    keepOnlySideEffects(lhs),
    keepOnlySideEffects(rhs),
    BooleanLiteral(true)))
```